Sources/Workspace/UserToolchain.swift

```diff
@@ -365,58 +385,93 @@ public final class UserToolchain: Toolchain {
             }
         }
 
-        // Compute the path of directory containing the PackageDescription libraries.
-        var pdLibDir = ToolchainConfiguration.libDir(forBinDir: binDir)
+        let customSwiftPMLibrariesLocation = try Self.deriveCustomSwiftPMLibrariesLocation(swiftCompilerPath: swiftCompilerPath, destination: destination)
+
+        let xctestPath = Self.deriveXCTestPath()
 
+        self.configuration = .init(
+            swiftCompilerPath: swiftCompilers.manifest,
+            swiftCompilerFlags: self.extraSwiftCFlags,
+            swiftPMLibrariesLocation: customSwiftPMLibrariesLocation,
+            sdkRootPath: self.destination.sdk,
+            xctestPath: xctestPath
+        )
+    }
+
+    private static func deriveCustomSwiftPMLibrariesLocation(
+        swiftCompilerPath: AbsolutePath,
+        destination: Destination
+    ) throws -> ToolchainConfiguration.SwiftPMLibrariesLocation? {
         // Look for an override in the env.
-        if let pdLibDirEnvStr = ProcessEnv.vars["SWIFTPM_PD_LIBS"] {
+        if let pathEnvVariable = ProcessEnv.vars["SWIFTPM_CUSTOM_LIBS_DIR"] ?? ProcessEnv.vars["SWIFTPM_PD_LIBS"] {
+            if ProcessEnv.vars["SWIFTPM_PD_LIBS"] != nil {
+                print("SWIFTPM_PD_LIBS was deprecated in favor of SWIFTPM_CUSTOM_LIBS_DIR")
+            }
```

Comment:
@abertelrud @neonichu please note ^^, I think this new name makes more sense. wdyt?

Reply:
I agree, this is clearer.